pkg/syncer/mutators/deployment.go

```diff
@@ -0,0 +1,247 @@
+package mutators
+
+import (
+	"net"
+	"net/url"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	utilspointer "k8s.io/utils/pointer"
+)
+
+type DeploymentMutator struct {
```

Comment:
Hmm, what happened to the idea of mutators being a function and not a type? Fine to special-case instantiation but that doesn't require special-casing registration or use e.g.

```golang
deploymentMutator := NewDeploymentMutator(...)
c.registerMutator(deploymentMutator.GVR, deploymentMutator.Mutate)
c.registerMutator(otherGVR, func(u *unstructured.Unstructured) error {
   // Do my thing
})
```

Reply:
Changed it, please take a look..